src/Rule/EmailHandler.php

```diff
@@ -67,7 +70,7 @@ public function validate(mixed $value, object $rule, ValidationContext $context)
                     $value
                 ));
                 if ($valid && $rule->isCheckDNS()) {
-                    $valid = checkdnsrr($matches['domain'] . '.') || checkdnsrr($matches['domain'] . '.', 'A');
```

Comment:
Are you sure that it is not need? May be this need for several edge cases?

Reply:
Yes, I tested it. Concatenation with dot is needed for older PHP versions. See https://www.php.net/manual/en/function.checkdnsrr.php. This advice is 6 years old. Seems like with PHP 8 this behavior was fixed and no longer actual. 

As for "A" record - from the variety of test cases not a single one fails and I can't find a new one. Suggest to remove it until it's found by us or reported by users. Otherwise it's a premature complication for unknown reasons. Let me know if you know it. That's why it's better to add such tests right away.

Reply:
Reminds me of https://github.com/yiisoft/rate-limiter/pull/25#discussion_r986981538. :slightly_smiling_face:

Reply:
Discussed in group chat, decided to leave it like that.